pkg/os/volume/api.go

```diff
@@ -268,6 +277,96 @@ func getTarget(mount string) (string, error) {
 	return volumeString, nil
 }
 
+// GetVolumeIDFromTargetPath returns the volume id of a given target path.
+func (VolumeAPI) GetClosestVolumeIDFromTargetPath(targetPath string) (string, error) {
+	volumeString, err := findClosestVolume(targetPath)
+
+	if err != nil {
+		return "", fmt.Errorf("error getting the closest volume for the path=%s, err=%v", targetPath, err)
+	}
+
+	return volumeString, nil
+}
+
+// findClosestVolume finds the closest volume id for a given target path
+// by following symlinks and moving up in the filesystem, if after moving up in the filesystem
+// we get to a DriveLetter then the volume corresponding to this drive letter is returned instead.
+func findClosestVolume(path string) (string, error) {
+	candidatePath := path
+
+	// run in a bounded loop to avoid doing an infinite loop
+	// while trying to follow symlinks
+	for i := 0; i < 256; i += 1 {
```

Comment:
I think 256 is just an arbitrary number we pick, right?
if you have some references, pls add it here. 
Also add this to comment, we only follow up to 256?

Reply:
Added comments about the why